### PR TITLE
Add tags argument to resource/aws_aws_organizations_organizational_unit (#15476)

### LINKED ISSUE #15476: tags attribute support for aws_organizations_organizational_unit
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add tags attribute support for aws_organizations_organizational_unit

### New or Affected Resource(s)

* aws_organizations_organizational_unit

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* https://docs.aws.amazon.com/organizations/latest/APIReference/API_CreateOrganizationalUnit.html

### PR DESCRIPTION
**Pull request by ryno75**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15476 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
ENHANCEMENTS

* resource/aws_organizations_organizational_unit: Add `tags` argument ([#15476](https://github.com/terraform-providers/terraform-provider-aws/issues/15476))
```

Output from acceptance testing:

```
❯ make testacc TESTARGS='-run=TestAccAWSOrganizations/OrganizationalUnit'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSOrganizations/OrganizationalUnit -timeout 120m
=== RUN   TestAccAWSOrganizations_serial
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnit
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnit/Tags
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnit/basic
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnit/Name
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnits
=== RUN   TestAccAWSOrganizations_serial/OrganizationalUnits/DataSource
=== PAUSE TestAccAWSOrganizations_serial/OrganizationalUnits/DataSource
=== CONT  TestAccAWSOrganizations_serial/OrganizationalUnits/DataSource
--- PASS: TestAccAWSOrganizations_serial (275.65s)
    --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnit (233.46s)
        --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnit/Tags (110.02s)
        --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnit/basic (46.11s)
        --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnit/Name (77.33s)
    --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnits (0.00s)
        --- PASS: TestAccAWSOrganizations_serial/OrganizationalUnits/DataSource (42.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	277.660s
```

### CODE REVIEW

**Comment by dzoeteman** [comment]
Could you add the tags to the Create action? This would allow the use of `aws:RequestTag` condition for example on IAM policies for terraform.

**Comment by ryno75** [comment]
> Could you add the tags to the Create action? This would allow the use of `aws:RequestTag` condition for example on IAM policies for terraform.

Not sure I understand. The tags are part of the Create action under this PR...
https://github.com/hashicorp/terraform-provider-aws/blob/6ef6001ffc7df938c0a21f7e2afad0e531bd47b4/aws/resource_aws_organizations_organizational_unit.go#L112-L116

Are you referring to something else?

**Review by anGie44 — COMMENTED** [review]
Hi @ryno75 , thank you for this PR! changes overall look great -- only note is that when setting `tags` on create, we can directly set those in the `CreateOrganizationalUnitInput` struct rather than call the `UpdateTags`, tho that certainly works as well. Additionally, could you please add a `.changelog` w/an entry similar to https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#resource-and-provider-enhancements ?

File: aws/resource_aws_organizations_organizational_unit.go [hunk 1]
```diff
@@ -99,19 +101,27 @@ func resourceAwsOrganizationsOrganizationalUnitCreate(d *schema.ResourceData, me
...
-		return fmt.Errorf("Error creating organizational unit: %s", err)
+		return fmt.Errorf("Error creating Organizational Unit: %s", err)
 	}
 	log.Printf("[DEBUG] Organizational Unit create response: %#v", resp)
 
 	// Store the ID
 	ouId := resp.OrganizationalUnit.Id
 	d.SetId(*ouId)
 
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
```
Comment (anGie44) [inline aws/resource_aws_organizations_organizational_unit.go:112]: we could alternatively add tags w/in the CreateInput struct above e.g.
```suggestion
	// Create the organizational unit
	createOpts := &organizations.CreateOrganizationalUnitInput{
		Name:     aws.String(d.Get("name").(string)),
		ParentId: aws.String(d.Get("parent_id").(string)),
		Tags:     keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().OrganizationsTags(),
	}
```

**Comment by anGie44** [comment]
Hi @ryno75, thank you again for this PR! Since time has passed since my last review, I've continued your work in #18861 to get the PR reviewed by another maintainer and into a future provider release, it includes your commits and addressing the comment above.

**Pull request closed without merging**